presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java

```diff
@@ -308,6 +308,7 @@ private int readAllNulls(int[] positions, int positionCount)
             }
             else {
                 outputPositionCount = positionCount;
+                outputPositions = positions;
```

Comment:
@bhhari What was the problem? Was it that `getReadPositions` returned junk? If so, shouldn't `if` branch also populate `outputPositions`?

Also, `outputPositions = positions` needs to be followed by `outputPositionsReadOnly = true;`

Reply:
@mbasmanova added the readOnly. whats happening here is that with IS_NULL filter then we are assigning only the position count. The "if" case should be covered by the line 245

```
if (!(nullsAllowed && nonNullsAllowed) || nullsFilter != null || listFilter != null) {
            outputPositions = ensureCapacity(outputPositions, positionCount);
        }
        else {
            outputPositions = positions;
            outputPositionsReadOnly = true;
        }
```

Reply:
@bhhari The `if` case is `if (nullsFilter != null) {`; the code above will allocate `outputPositions`, but it won't populate it with correct values. I think you need to add logic to populate `outputPositions` in the `if` block. Am I wrong?

Reply:
you are right 👍